recipes/Libri-Light/self-supervised-learning/make_librilight_csv.py

```diff
@@ -0,0 +1,168 @@
+""" Create speebrain csv files for the Libri-Light dataset for self-supervised-learning
+
+1. Download the Libri-Light dataset through the toolkit in the Libri-Light github repo
+    "https://github.com/facebookresearch/libri-light/"
+
+2. Use the data_preparation/cut_by_vad.py script of the Libri-Light repo to do the vad. For example,
+    "python cut_by_vad.py --input_dir path_to_Libri-Light_small --output_dir Libri-Light_small_vad --target_len_sec 20"
+
+3. Use the vad output_dir in step 2 as the input_dir in this step to generate the csv file. For example,
+    "python make_librilight_csv.py --input_dir Libri-Light_small_vad --output_dir results --max_length 20 --n_processes 128" to generate the train.csv for the SSL pretraining
+
+4. Now, you can use the generated train.csv in the output_dir in step 3 as the "train_csv" in any SpeechBrain SSL pre-training yaml file
+
+Authors
+ * Shucong Zhang 2024
+ * Titouan Parcollet 2024
+"""
+
+import argparse
+import csv
+import multiprocessing
+import os
+import pathlib
+from pathlib import Path
+
+import torchaudio
+import tqdm
+
+
+def make_csv_for_each(
+    subpath_1_csv_file_folder_max_length: tuple,
+) -> None:
+    """Prepare the csv files for each subfolder of the Libri-Light splits
+
+    Arguments
+    ---------
+    subpath_1_csv_file_folder_max_length : tuple
+        A (subfolder_path, csv_output_path, max_utt_length) tuple
+    """
+    subpath_1, csv_file_folder, max_length = (
+        subpath_1_csv_file_folder_max_length
+    )
+    for i, flac_file in enumerate(subpath_1.glob("**/*.flac")):
+        flac_file_name = flac_file.stem
+        waveform, sample_rate = torchaudio.load(str(flac_file))
```

Comment:
I think it would be better to instead use the `read_audio_info` (see: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/dataio/dataio.py#L166-L221)